File: S1MCPServer/Models/Acknowledgment.cs
using System.Text.Json.Serialization;

namespace S1MCPServer.Models;

/// <summary>
/// Represents an acknowledgment message from the client.
/// </summary>
public class Acknowledgment
{
    /// <summary>
    /// Request ID that this acknowledgment corresponds to.
    /// </summary>
    [JsonPropertyName("id")]
    public int Id { get; set; }

    /// <summary>
    /// Acknowledgment status.
    /// </summary>
    [JsonPropertyName("status")]
    public string Status { get; set; } = "received";
}

File: S1MCPClient/src/models/acknowledgment.py
"""Acknowledgment model for JSON-RPC communication."""

from pydantic import BaseModel, Field


class Acknowledgment(BaseModel):
    """Acknowledgment model for confirming receipt of responses."""
    
    id: int = Field(..., description="Request ID that this acknowledgment corresponds to")
    status: str = Field(default="received", description="Acknowledgment status")
    
    class Config:
        """Pydantic configuration."""
        json_schema_extra = {
            "example": {
                "id": 1,
                "status": "received"
            }
        }

